Source/Core/CoplanarPolygonGeometry.js

```diff
@@ -128,18 +129,27 @@ function createGeometryFromPolygon(
     flatPositions[positionIndex++] = position.z;
 
     if (vertexFormat.st) {
-      const p = Matrix3.multiplyByVector(
-        textureMatrix,
-        position,
-        scratchPosition
-      );
-      const st = projectPointTo2D(p, stScratch);
-      Cartesian2.subtract(st, stOrigin, st);
+      if (
+        hardcodedTextureCoordinates &&
+        hardcodedTextureCoordinates.positions.length === length
+      ) {
+        textureCoordinates[i * 2] = hardcodedTextureCoordinates.positions[i].x;
```

Comment:
```suggestion
        textureCoordinates[stIndex++] = hardcodedTextureCoordinates.positions[i].x;
```

Would this also just work? It'd be better if we could just use the variable in both cases.

Reply:
Agreed, will commit in batch.